cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -73,6 +73,21 @@ func CreateInitKubeConfigFiles(outDir string, cfg *kubeadmapi.MasterConfiguratio
 	)
 }
 
+// CreateJoinMasterKubeConfigFiles will create and write to disk the kubeconfig files required by kubeadm
+// join --master workflow, plus the admin kubeconfig file to be deployed on the new master; the
+// kubelet.conf fil must not be created when joining master nodes because it will be created and signed by
+// the kubelet TLS bootstrap process.
+// If any kubeconfig files already exists, it used only if evaluated equal; otherwise an error is returned.
+func CreateJoinMasterKubeConfigFiles(outDir string, cfg *kubeadmapi.MasterConfiguration) error {
+	return createKubeConfigFiles(
+		outDir,
+		cfg,
+		kubeadmconstants.AdminKubeConfigFileName,
```

Comment:
note: this creates `n` different root users to the cluster. Not sure that is desirable from a security standpoint, but it's probably something we should do for a good UX

Reply:
I agree.
As a workaround the user can always copy the admin.conf file from the bootstrap master to the joining master (so there will be only 1 root user, distributed on all the masters)